oneflow/core/framework/dtype.cpp

```
@@ -0,0 +1,99 @@
+/*
+Copyright 2020 The OneFlow Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#include "oneflow/core/common/util.h"
+#include "oneflow/core/framework/dtype.h"
+
+namespace oneflow {
+
+Maybe<DType> DType::GetDTypeByDataType(const DataType& data_type) {
+  switch (data_type) {
+#define MAKE_DATA_TYPE_OBJ(data_type) \
+  case DataType::k##data_type: {      \
+    return data_type();               \
+  }
+    OF_PP_FOR_EACH_TUPLE(MAKE_DATA_TYPE_OBJ, DTYPE_SEQ)
+#undef MAKE_DATA_TYPE_OBJ
+    default: {
+      static std::shared_ptr<DType> invalid_dtype = std::make_shared<DType>(
+          DataType::kInvalidDataType, "oneflow.invalid_data_type", false, false, false);
+      return invalid_dtype;
```

OF_UNIMPLEMENTED();
到这里已经出错了，应该fail fast